### PR TITLE
Remove validate current step checkbox and show validation error messages

### PR DESCRIPTION
**Pull request by melohagan**
## Description
1. Remove validate current form step validation (it's always checked)
2. Display the validation error messages instead of generic 'Bad Request'

Addresses: 
- https://github.com/Budibase/budibase/issues/6820
- https://github.com/Budibase/budibase/issues/6356

## Screenshots
**Validation checkbox removed**
Screenshot 2022-07-20 at 13 58 46

**Validation error messages**
Screenshot 2022-07-20 at 13 59 29

### CODE REVIEW

**Comment by melohagan** [comment]
@aptkingston I thought the single giant error notification would look weird because the toasts are designed to be single lines. It also encourages the user to acknowledge each error. 

These errors are also specific to validation, so ideally only the app developer should be seeing them, and then fix it by adding the validation action to their form so that invalid submissions can't take place. 

Maybe I could combine the old and new code. So if it's a validation error then stack. If it's any other type of error then clear the stack - as was the case before. 

Happy to change it in whatever way suits!

**Comment by aptkingston** [comment]
No need to change! Happy to leave it as is especially since you'd already considered that option :+1:

File: packages/client/src/components/overlay/NotificationDisplay.svelte [hunk 1]
```diff
@@ -39,7 +41,8 @@
     display: flex;
     flex-direction: column;
     justify-content: flex-start;
-    align-items: center;
+    align-items: end;
```
Comment (aptkingston) [inline packages/client/src/components/overlay/NotificationDisplay.svelte:44]: After testing it out myself, this is the only thing I think we need to change. Aligning notifications to the right looks strange to me, as it even means single notifications aren't actually centered in the screen. I think we should change this back to `center`.

Comment (melohagan) [reply packages/client/src/components/overlay/NotificationDisplay.svelte]: Changed back to center align with fixed toast widths.

File: packages/client/src/stores/notification.js [hunk 2]
```diff
@@ -37,26 +39,32 @@ const createNotificationStore = () => {
...
+          dismissable: !autoDismiss,
+          delay: get(store) != null,
+        },
+      ]
     })
-    clearTimeout(timeout)
     if (autoDismiss) {
-      timeout = setTimeout(() => {
-        store.set(null)
+      const timeoutId = setTimeout(() => {
```
Comment (aptkingston) [inline packages/client/src/stores/notification.js:57]: As above, we don't need to worry about storing a reference. We can just call `setTimeout`.

File: packages/client/src/stores/notification.js [hunk 3]
```diff
@@ -37,26 +39,32 @@ const createNotificationStore = () => {
...
+        dismiss(_id)
       }, NOTIFICATION_TIMEOUT)
+      timeoutIds.add(timeoutId)
```
Comment (aptkingston) [inline packages/client/src/stores/notification.js:60]: As above, we can just remove this line entirely.

**Review by aptkingston — COMMENTED** [review]
Tested the changes myself there Mel, and while I really like the fixed width notifications, I experienced some problems due to the use of `timeoutIds` and the store subscription callback. Removing all of that logic both tidies up the logic and fixes the problems thankfully!

**Comment by melohagan** [comment]
That's what I get for copy/pasting - thanks for having look!

I've removed the timeoutIds and tested it. Seems to work from what I can see.

**Review by aptkingston — APPROVED** [review]
LGTM!

**Pull request merged**